src/main/java/com/marklogic/client/spring/BasicConfig.java

```diff
@@ -66,7 +66,7 @@ protected String buildContentDatabaseName(String mlAppName) {
         return mlAppName + "-content";
     }
 
-    @Bean
+    @Bean(name = "default")

```

Comment:
I am 99% sure that the bean already has a name of "databaseClientProvider". 

You should be able to define a 2nd Bean method that returns a DatabaseClientProvider, and that method just needs a different name. Spring by default will use the method name as the bean name. 

Clients must then specify which bean they want via a Qualifier annotation.